### ISSUE TITLE
Fuzzer

### ISSUE THREAD

**Issue by PtrMan** [opennars/opennars#276]
Write a fuzzer and let NARS run with it for hours or days.

**Comment by TonyLo1** [comment]
Property based testing can achieve the same thing in a more structured way.

**Comment by PtrMan** [comment]
I am doubtful - how should this work if the whole NAR is under test?

**Comment by TonyLo1** [comment]
It's just a class that needs testing same as any other class - you just need to create the constraining predicates to generate the test cases from - at the NAR level the predictates will relate to task creation. There will be predicates for sentences, statements, terms, etc. then random but valid task test cases can be created automatically with good coverage if a sufficiently large number of test cases are generated.

**Comment by PtrMan** [comment]
So it's the same

**Comment by TonyLo1** [comment]
I agree that it is similar. The difference is that with property based testing, 'shrinking' will find the minimal test case that causes the failure.

**Comment by PtrMan** [comment]
Oh this makes sense!
